WIP: LODCM Branching Specification

teddyrendahl opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
## Description
<!--- Describe your changes in detail -->
The `lightpath` requires that an optic report all possible beamlines that may branch off the line it is on. Unfortunately, since both `LODCM` are on the `HXD` branch which has **all** the HXR beamlines it must report them. The `main_line` argument has been changed to accept a `list`. For consistency sake I changed the `mono_line` to use a list as well. This last choice is debatable. 

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Have the lightpath report accurately

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
* I tried this with the `lightpath` and it correctly identified the LODCM as a blocking device for the `HXR` hutches. 
* Existing tests pass unmodified

## Where Has This Been Documented?
<!--  Include where the changes made have been documented. -->
<!--  This can simply be  a comment in the code or updating a docstring -->
Updated docstrings

ZLLentz commented:
Are you saying that the `lightpath` needs each `LODCM` to report all possible destinations, even if the `LODCM` doesn't directly feed into them? Is this true for all devices? Seems like a strict requirement, maybe I misunderstand.

teddyrendahl commented:
I wrote a long response to why, but after thinking about it more I might just patch lightpath. Posting below for posterity.

teddyrendahl commented:
We changed how the `lightpath` worked since `skywalker`. The logic is as following:

1. Traverse the beamline, remember any optics we see
2. Stop when we find a device with a different beamline then the one we are on.
3. Check that all optics we have seen up to this point are configured to deliver to this new beamline. Otherwise, mark them as blocking.
4. Clear our cache of seen optics and repeat

This was done to deal with some complicated situations without giving the beamlines between every optic a unique name. This is a special pain for the few devices in between XRT M1H / XRT M2H.

Now, the way we define our beamlines is that the HXD line starts in the FEE and travels all the way to the XCS LODCM. That means when we split off of the main line all the way down at CXI, we still have both LODCMs in our "cache" of seen optics and want to see they point to CXI.
